[FIX] Batch fixes and improvements

dolfinus opened the pull request:
- Fixed requests errors:
- -   using `extract_arguments` for GET /api/model/:id request with no body
- -   returning `werkzeug.exceptions.BadRequest` whle passing body for GET requests
- -   ```datetime.datetime is not JSON serializable``` while GET date fields
- -   Removed magic of parsing domain value for GET /api/model

- Fixed using `application/www-form-urlencoded` instead of `application/json` for request data
- Return record data instead on non-JSON messages for PUT and PATCH requests
- Return no data for DELETE request
- Id and model name checks are wrappers now instead of copy-paste
- Using exact routes for each method
- Updated documentation for methods usage

dolfinus committed:
[FIX] Batch fixes and improvements
Fixed requests errors:
  using `extract_arguments` for GET /api/model/:id request with no body
  returning `werkzeug.exceptions.BadRequest` whle passing body for GET requests
  ```datetime.datetime is not JSON serializable``` while GET date fields
  Removed magic of parsing domain value for GET /api/model
Fixed using `application/www-form-urlencoded` instead of `application/json`
Return record data instead on non-JSON messages for PUT and PATCH requests
Return no data for DELETE request
Id and model name checks are wrappers now instead of copy-paste
Using exact routes for each method
Updated documentation for methods usage

dolfinus commented:
This pull request also fixes [issue with Access Token request](https://github.com/ajepe/odoo-addons/issues/13).

dolfinus committed:
Merge branch '12.0' into 12.0

dolfinus committed:
Move domain value parsing to function

dolfinus committed:
Divide Odoo tulpe-ids to int and string part
Like country_id: [1, 'United States'] => country_id: 1, country: 'United States'
No more validations of non-int ids at client side

dolfinus committed:
Fixed domain parsing result empty

dolfinus committed:
Parse order as list, not str

dolfinus committed:
Do not send Body in GET requests
Search method now uses query params, so no body required

ajepe commented:
There are series of conflict with the existing code base, would have love to merge this. but will find a way around it when am a bit free.

Commit:
Fix order arg parsing
You can use both str and list as order in search query
Some docs fixes

Commit:
Fix put args str type values
While you pass form with stage_id=4 it become int instead of str

Commit:
Fix usage of invalid_request in Token controller

Commit:
Use POST request for credentials to generate token
Do not pass them in headers, because proxies can read headers
Do not use query string for passing them because everybody can see it

Commit:
Get rid of simpejson package
Json is installed with Odoo by default
